### ISSUE TITLE
ERROR: lazy loading failed for package ‘spData’

### ISSUE THREAD

**Issue by lassancejm** [Nowosad/spData#51]
Hi, 

I am encountering problems when trying to install the package (either from CRAN or github) and would be glad if you could help me getting a hint of what is causing the issues. 

Thanks!

Here is the error message:
```
* installing *source* package ‘spData’ ...
** package ‘spData’ successfully unpacked and MD5 sums checked
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
sh: line 1: 158092 Segmentation fault      (core dumped) R_TESTS= '/n/helmod/apps/centos7/Core/R_core/4.0.2-fasrc01/lib64/R/bin/R' --no-save --no-restore --no-echo 2>&1 < '/tmp/RtmpT2zOg4/file2697c42ed6f10'

 *** caught segfault ***
address 0x20, cause 'memory not mapped'

Traceback:
 1: Module(module, mustStart = TRUE, where = env)
 2: doTryCatch(return(expr), name, parentenv, handler)
 3: tryCatchOne(expr, names, parentenv, handlers[[1L]])
 4: tryCatchList(expr, classes, parentenv, handlers)
 5: tryCatch(Module(module, mustStart = TRUE, where = env), error = function(e) e)
 6: loadModule(module = "spmod", what = TRUE, env = ns, loadNow = TRUE)
 7: (function (ns) loadModule(module = "spmod", what = TRUE, env = ns, loadNow = TRUE))(<environment>)
 8: doTryCatch(return(expr), name, parentenv, handler)
 9: tryCatchOne(expr, names, parentenv, handlers[[1L]])
10: tryCatchList(expr, classes, parentenv, handlers)
11: tryCatch((function (ns) loadModule(module = "spmod", what = TRUE, env = ns, loadNow = TRUE))(<environment>),     error = function(e) e)
12: eval(substitute(tryCatch(FUN(WHERE), error = function(e) e),     list(FUN = f, WHERE = where)), where)
13: eval(substitute(tryCatch(FUN(WHERE), error = function(e) e),     list(FUN = f, WHERE = where)), where)
14: .doLoadActions(where, attach)
15: methods::cacheMetaData(ns, TRUE, ns)
16: loadNamespace(j <- imp[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
17: asNamespace(ns)
18: asNamespace(ns)
19: namespaceImportFrom(self, asNamespace(ns), vars, from = from)
20: namespaceImportClasses(ns, loadNamespace(j <- imp[[1L]], c(lib.loc,     .libPaths()), versionCheck = vI[[j]]), imp[[2L]], from = package)
21: loadNamespace(package = package, lib.loc = lib.loc, keep.source = keep.source,     keep.parse.data = keep.parse.data, partial = TRUE)
22: withCallingHandlers(expr, packageStartupMessage = function(c) tryInvokeRestart("muffleMessage"))
23: suppressPackageStartupMessages(loadNamespace(package = package,     lib.loc = lib.loc, keep.source = keep.source, keep.parse.data = keep.parse.data,     partial = TRUE))
24: code2LazyLoadDB(package, lib.loc = lib.loc, keep.source = keep.source,     keep.parse.data = keep.parse.data, compress = compress, set.install.dir = set.install.dir)
25: tools:::makeLazyLoading("spData", "/n/holylfs03/LABS/hoekstra_lab/Users/lassance/apps/R/00LOCK-spData/00new",     keep.source = FALSE, keep.parse.data = FALSE, set.install.dir = "/n/holylfs03/LABS/hoekstra_lab/Users/lassance/apps/R/spData")
An irrecoverable exception occurred. R is aborting now ...
ERROR: lazy loading failed for package ‘spData’
* removing ‘/n/holylfs03/LABS/hoekstra_lab/Users/lassance/apps/R/spData’

The downloaded source packages are in
	‘/tmp/RtmpTBn5To/downloaded_packages’
Warning message:
In install.packages("spData") :
  installation of package ‘spData’ had non-zero exit status
```

Full details about the setup:
```
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /usr/lib64/libblas.so.3.4.2
LAPACK: /usr/lib64/liblapack.so.3.4.2

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.0.2 tools_4.0.2  
```

**Comment by Robinlovelace** [comment]
Interesting. Do you know if this is reproducible in a Docker container, e.g. https://github.com/rstudio/rstudio/blob/master/docker/jenkins/Dockerfile.centos7-x86_64 ?

And have you tried on different platforms/R versions?

Overall I'm not sure what's going on, thanks for opening the issue and hope it can be resolved!

**Comment by lassancejm** [comment]
OK, I figure it out, and it is thanks to your suggestion to try installing on the different platform. 

So, I tried to install the package on my Mac and it worked without any problem. What I noticed though, is this in the prompt ```also installing the dependencies ‘Rcpp’, ‘sp’, ‘raster’```. This message made me realize that the error may not have anything to do with `spData` but with any of those dependencies which I had not realized existed.

After re-installing the dependencies, the installation of `spData` completed successfully. 

Problem solved then, thanks for your time!
